src/urllib3/connection.py

```diff
@@ -217,7 +217,7 @@ def request(self, method, url, body=None, headers=None):
         headers = HTTPHeaderDict(headers if headers is not None else {})
         if "user-agent" not in headers:
             headers["User-Agent"] = _get_default_user_agent()
-        elif headers["user-agent"] == SUPPRESS_USER_AGENT:
+        elif SKIP_USER_AGENT in headers.get_all("user-agent"):
```

Comment:
Maybe we can test that?
```python
headers.add("User-Agent", "Normal")
headers.add("User-Agent", SKIP_USER_AGENT)
assert skipped
```

I'll try to add a test later :)

Reply:
Good idea!